packages/shared/scope-hoisting/src/link.js

```diff
@@ -217,35 +217,33 @@ export function link({
     if (mod.meta.isCommonJS && originalName === 'default') {
       let name = getName(mod, '$interop$default');
       if (!path.scope.getBinding(name)) {
-        // Hoist to the nearest path with the same scope as the exports is declared in
-        let binding = path.scope.getBinding(
-          assertString(mod.meta.exportsIdentifier),
+        let binding = nullthrows(
+          path.scope.getBinding(
+            bundle.hasAsset(mod)
+              ? assertString(mod.meta.exportsIdentifier)
+              : // If this bundle doesn't have the asset, use the binding for
+                // the `parcelRequire`d init function.
+                getIdentifier(mod, 'init').name,
```

Comment:
```suggestion
                getName(mod, 'init'),
```
The identifier is actually already in the `node` parameter, but it's either an Identifier or a CallExpression. Not really cleaner.